lib/array_intersection.rb

```diff
@@ -1,6 +1,43 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+
+# Brute Force
+# Time complexity: O(n) * O(m), where n and m are the number of elements in each respective array
+# Space complexity: O(n) * O(m), where n and m are the number of elements in each respective array
 def intersection(array1, array2)
-  raise NotImplementedError
+  return_array = []
+
+  unless array1.nil? || array2.nil? 
+    array1.each do |i|
+      array2.each do |j|
+        if i == j
+          return_array << i  
+        end
+      end
+    end
+  end
+  return return_array
 end
+
+## Attempt 1 at Hash Table
+# def intersection(array1, array2)
+#   return_array = []
+#   hash_table = {}
+
+#   unless array1.nil? || array2.nil?
+#     if array1.count < array2.count
```

Comment:
This `if` statement would need an `else` to add the elements from array2 into the hash.  However you don't really need an if here, you could just put all elements from array1 into the hash and then loop through array2 in every case.